raiden/transfer/channel.py

```diff
@@ -577,16 +633,27 @@ def get_current_balanceproof(end_state: NettingChannelEndState) -> BalanceProofD
         locksroot = balance_proof.locksroot
         nonce = balance_proof.nonce
         transferred_amount = balance_proof.transferred_amount
+        locked_amount = get_amount_locked(end_state)
     else:
         locksroot = EMPTY_MERKLE_ROOT
         nonce = 0
         transferred_amount = 0
+        locked_amount = 0
 
-    return (locksroot, nonce, transferred_amount)
+    return (locksroot, nonce, transferred_amount, locked_amount)
 
 
 def get_distributable(sender: NettingChannelEndState, receiver: NettingChannelEndState) -> int:
-    return get_balance(sender, receiver) - get_amount_locked(sender)
+    _, _, transferred_amount, locked_amount = get_current_balanceproof(sender)
+
+    distributable = get_balance(sender, receiver) - get_amount_locked(sender)
+
+    overflow_limit = max(
+        UINT256_MAX - transferred_amount - locked_amount,
+        0,
+    )
+
+    return min(overflow_limit, distributable)
```

Comment:
Can we add a comment in this function explaining what is happening here? At first sight it is a bit cryptic.
E.g.:
`We need to limit the transferred_amount + locked_amount to avoid overflows. The distributable amount itself should not be bigger under normal circumstances. This is an additional security check.`

Reply:
done